com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineMaterialUpgrader.cs

```diff
@@ -440,10 +441,75 @@ internal SpeedTreeBillboardUpgrader(string oldShaderName)
     }
     internal class SpeedTree8Upgrader : MaterialUpgrader
     {
+        private enum WindQuality
```

Comment:
there's a lot of duplicate code with HDRP, maybe some of it can be moved to render-pipelines.core ?

Reply:
Maybe I should move it to shadergraph instead? Since this upgrader is specifically for the ST8 shadergraph.

Reply:
Ah, SHadergraph doesn't have MaterialUpgrader defined because that's defined in core.

Reply:
shadergaph has a dependency on core rp though so you should still be able to access it